8ofproject/2ch-Browser-iOS
# Add webm support

### Background
WEBM support was down for a while, and it's frustrating given that most video content on 2ch is in that exact format. Since iOS has no native support for that format, I had to use [VLCKit](https://code.videolan.org/videolan/VLCKit)

### What has been done
1. Updated target to deploy target to iOS 9 (needed 9.0 for safe area support.)
2. Add VLCKit pod 
3. Integrated  Swift into the project
4. Created VLCPlayer view controller
5. Add Icons for the player view
6. Update readme file to reflect the new minimal iOS version

### How to test
0. Proofread the code (might have grammatical errors)
1. Start the app
2. Open a thread that has WebM files
3. Tap on any WebM file
4. Make sure the video starts muted
5. Test that the controls are working properly
6. Make sure that the play/pause and mute/unmute buttons have opposite action relative to the player state. (ex. should be pause while video playing)

### TODO (sometimes in the future)
- Player controls have a bit of delay on tap
- There is a bug that will not let the video to play from the start once the playback is finished.
- Update deprecated code.
- Player controls are not hiding.
- No error is presented when the URL to WebM is broken
